packages/typir/src/features/conversion.ts

```diff
@@ -136,20 +136,16 @@ export class DefaultTypeConversion implements TypeConversion {
             edge.mode = mode;
         }
 
-        // check, that the new edges did not introduce cycles
-        this.checkForCycles(mode);
-    }
-
-    protected checkForCycles(mode: ConversionModeForSpecification): void {
         if (mode === 'IMPLICIT_EXPLICIT') {
-            this.checkForCyclesLogic(mode);
-        } else {
-            // all other modes allow cycles
+            /* check that the new edges did not introduce cycles
+             * if it did, the from node will be reachable via a cycle path
+             */
+            const hasIntroducedCycle = this.existsEdgePath(from, from, mode);
+            if (hasIntroducedCycle) {
+                throw new Error('Adding the conversion from ' + from.identifier + ' to ' + to.identifier + ' has introduced a cycle ot the type graph.');
```

Comment:
- Maybe adding the `mode` as well would make the error message even more clear?
- typo: "ot"

Reply:
Ah, good catch! Thanks, and thanks for reviewing!